snakemake/shell.py

```diff
@@ -132,10 +142,10 @@ def __new__(
             cmd,
             bufsize=-1,
             shell=True,
-            stdout=stdout,
             universal_newlines=iterable or None,
             close_fds=close_fds,
-            **cls._process_args
+            **cls.__get_output_streams(context.get("log"), iterable or read),
+            **cls._process_args,
```

Comment:
The open files that you get with `Log.streams()` should be closed below.

Reply:
They are closed by Popen, which has close_fds=True, though I guess if I close them manually then this feature might even work for Windows platform.